### PR TITLE
Increase default input size

### PR DESCRIPTION
**Pull request by runn1ng**
1b larger input size; correct output size in simulation.

### CODE REVIEW

**Comment by runn1ng** [comment]
The tests don't like the second change. I can either remove the commit or change the tests, what do you think.

File: utils.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 // baseline estimates, used to improve performance
 var TX_EMPTY_SIZE = 4 + 1 + 1 + 4
 var TX_INPUT_BASE = 32 + 4 + 1 + 4
-var TX_INPUT_PUBKEYHASH = 106
+var TX_INPUT_PUBKEYHASH = 107
```
Comment (dcousens) [inline utils.js]: This is variable.
I think the "average" range is going to be about 106-108,  but,  variable.

File: stats/index.js [hunk 2]
```diff
@@ -14,7 +14,7 @@ for (var j = 0; j < 100; ++j) {
   for (var i = 1; i < 4; ++i) {
     let utxos = Simulation.generateTxos(20 / i, min, max)
     let txos = Simulation.generateTxos(80 / i, min, max / 3)
-
+    txos.forEach(txo => txo.script.length = 34)
```
Comment (dcousens) [inline stats/index.js]: Any reason to specify this even though it is the default?

**Comment by dcousens** [comment]
For 100 random public keys, I got:
```
      1 106
     44 107
     55 108
```

Update the PR to 108, fix the fixtures, and we can merge :+1:

File: stats/index.js (hunk 2 above)
Comment (runn1ng) [inline stats/index.js]: because by default, script.length will be 106 or 300, in Simulation.generateTxos (L 38)

and that is wrong for *outputs*. BnB strategy counts bytelength of outputs (for cost); it counts them wrong.

**Comment by runn1ng** [comment]
Are you sure it can be 108? Isn't there some 1 byte overhead for length, that  is already in `TX_INPUT_BASE`?

from this answer
https://bitcoin.stackexchange.com/a/48280/1760

prevout is 32 + 4
sequence is 4
scriptsig length is 1 (that's in TX_INPUT_BASE already however)

scriptsig itself is 1 + 71/72 + 1 + 33, that gives 107 at max

scriptsig length should probably be in the script and not in TX_INPUT_BASE however. (And then I am not sure how you got to the 1 case with 106.)

File: stats/index.js (hunk 2 above)
Comment (runn1ng) [inline stats/index.js]: it might be better to just add some option to generateTxos? To let it know if it's inputs or outputs

Comment (runn1ng) [inline stats/index.js]: It's the same issue that I solved in the other simulation here
https://github.com/bitcoinjs/coinselect/pull/15/commits/96948ce6a6ae89f54850e78ff3b252076fbac6a2

(script.length cannot be empty, otherwise simulation thinks it's change and adds it back to utxos)

**Comment by dcousens** [comment]
@runn1ng oops!  That did include the varint,  so yes,  107.
Tests need fixing :+1: 

> (And then I am not sure how you got to the 1 case with 106.)

Because signatures are DER encoded,  and they don't pad zeroes out.
See https://github.com/bitcoinjs/bip66/blob/master/index.js#L68

File: stats/index.js (hunk 2 above)
Comment (runn1ng) [inline stats/index.js]: should I keep this here, or put it into a different PR? (it is in separate commit already)

**Comment by dcousens** [comment]
@runn1ng tests need fixing :+1:

File: stats/index.js (hunk 2 above)
Comment (dcousens) [inline stats/index.js]: maybe different PR,  I'm still not 100% why it is necessary,  apologies (no doubt my fault)

**Comment by runn1ng** [comment]
I am on it :D

**Comment by runn1ng** [comment]
Done. I have taken out the simulation output size commit, will make another PR

**Comment by runn1ng** [comment]
Hm. With the stats I made here, 106 is slightly more "popular".

https://gist.github.com/runn1ng/8e2881e5bb44e01748e46b3d1c549038

However. The average is 109.118 - there are a few 134 and 135-byte ones (together 8% of PKH inputs), and they bring the average to 109.1

So... I don't know. I don't want to bring probability here (hard to test:)). Maybe the default size is a mistake and it should be really a variable a added to the tests too?
